packages/playground/website/src/components/ensure-playground-site/ensure-playground-site-is-selected.tsx

```diff
@@ -41,6 +45,16 @@ export function EnsurePlaygroundSiteIsSelected({
 	const requestedSiteObject = useAppSelector((state) =>
 		selectSiteBySlug(state, requestedSiteSlug!)
 	);
+	const requestedClientInfo = useAppSelector(
+		(state) =>
+			requestedSiteSlug &&
+			selectClientBySiteSlug(state, requestedSiteSlug)
+	);
+	const [needMissingSitePromptForSlug, setNeedMissingSitePromptForSlug] =
+		useState<false | string>(false);
+
+	const promptIfSiteMissing =
+		url.searchParams.get('if-stored-site-missing') === 'prompt';
```

Comment:
Why do we need this extra query argument? Is there any harm in just prompting the user by default?

As a user, I prefer fewer query arguments to remember and type. Also, showing the modal is a reasonable default.

Reply:
Without that parameter, users will see the modal if they delete one of their Playgrounds and then visiting a stale `?slug=` link in from their browser history. The extra param ensures the modal is only visible when consent was the intention in the first place.